modules/gmake2/gmake2_cpp.lua

```diff
@@ -603,23 +600,25 @@
 
 	function cpp.fileFlags(cfg, file)
 		local fcfg = fileconfig.getconfig(file, cfg)
-		if fcfg and fcfg.flagsVariable then
-			return fcfg.flagsVariable
-		end
+		local flags = {}
 
-		if fcfg and fcfg.compileas then
-			if p.languages.isc(fcfg.compileas) then
-				return 'ALL_CFLAGS'
-			elseif p.languages.iscpp(fcfg.compileas) then
-				return 'ALL_CXXFLAGS'
-			end
+		if cfg.pchheader and not cfg.flags.NoPCH and (not fcfg or not fcfg.flags.NoPCH) then
+			table.insert(flags, "-include $(PCH_PLACEHOLDER)")
 		end
 
-		if path.iscfile(file.name) then
-			return 'ALL_CFLAGS'
+		if fcfg and fcfg.flagsVariable then
```

Comment:
I'm not entirely sure what `flagsVariable` is, why that would override the `$(ALL_*FLAGS)` flag, or how it's related to PCH files?

Reply:
This is the per-file arguments variable name.

With gmake2, if files have specific arguments, the generator will create a Makefile var `PERFILE_FLAGS_*` to group them.

As an example, You can have this:
`PERFILE_FLAGS_0 = $(ALL_CXXFLAGS) -mf16c`

then your file rule will be
`$(SILENT) $(CXX) $(PERFILE_FLAGS_0) $(FORCE_INCLUDE) -o "$@" -MF "$(@:%.o=%.d)" -c "$<"`
instead of
`$(SILENT) $(CXX) $(ALL_CXXFLAGS) $(FORCE_INCLUDE) -o "$@" -MF "$(@:%.o=%.d)" -c "$<"`

The way it is implemented is that `ALL_*FLAGS` is contained in `PERFILE_FLAGS_*` if necessary so it doesn't need to be added as well.